dev-client/src/screens/HomeScreen/components/SiteMapCallout.tsx

```diff
@@ -85,3 +60,51 @@ export const SiteMapCallout = ({sites, state, setState}: Props) => {
     </Mapbox.MarkerView>
   );
 };
+
+const CalloutChild = (coords: Coords, {sites, state, setState}: Props) => {
+  const closeCallout = useCallback(() => setState(noneCallout()), [setState]);
+
+  switch (state.kind) {
+    case 'site':
+      const site = getCalloutSite(state, sites);
+      if (!site) {
```

Comment:
comment: I made an arbitrary call here on functionality: if the user's selected site is not available in filtered data, the callout will be invisible until the filter includes it again. (The same happens for an empty list of cluster sites). We could also simply invoke the `closeCallout` function to clear the user's selection, but this approach felt more user-friendly.